src/main/features/claim/guards/claimEligibilityGuard.ts

```diff
@@ -0,0 +1,47 @@
+import * as express from 'express'
+
+import { GuardFactory } from 'response/guards/guardFactory'
+
+import { Draft } from '@hmcts/draft-store-client'
+import { DraftClaim } from 'drafts/models/draftClaim'
+import { DraftService } from 'services/draftService'
+
+import { CookieEligibilityStore } from 'eligibility/store'
+
+import { Paths as EligibilityPaths } from 'eligibility/paths'
+
+const eligibilityStore = new CookieEligibilityStore()
+
+export class ClaimEligibilityGuard {
+  /**
+   * Makes sure that eligibility check has passed prior accessing protected pages. The eligibility is assessed by
+   * checking whether draft has been marked as eligible or whether eligible cookie exists. If none of the conditions
+   * is met then user is redirected to eligibility index page, otherwise request is let through as is.
+   *
+   * @returns {express.RequestHandler} - request handler middleware
+   */
+  static check (): express.RequestHandler {
```

Comment:
We tend to name this method `requestHandler` because it returns an Express request handler.

Reply:
You are right, renamed it.